plugins/subcommandPermCheck.ts

```diff
@@ -0,0 +1,110 @@
+//@ts-nocheck
+/**
+ * @plugin
+ * Inspired by the plugin "requirePermission" created by Benzo-Fury & needhamgary, this plugin will set permissions for specific subcommands without manually writing it into the code.
+ *
+ * @author @Peter-MJ-Parker [<@371759410009341952>]
+ * @version 1.0
+ * @example
+ * ```ts
+ * import { subcommandPermCheck } from "../plugins/subcommandPerms";
+ * import { commandModule, CommandType } from "@sern/handler";
+ * import { PermissionFlagBits } from "discord.js";
+ * export default commandModule({
+ *  type: CommandType.Slash,
+ *  plugins: [
+ *   subcommandPermCheck(
+ *    [
+ *      { name: "string", perms: [PermissionFlagBits.Administrator] },
+ *      { name: "number", perms: [PermissionFlagBits.SendMessages, PermissionFlagBits.UseVAD] }
+ *    ],
+ *   all: true, //all = Require the member to have all perms stated or at least one?
+ *   "OPTIONAL - respond to user with this message or default."
+ *   )
+ *  ],
+ *  execute: ({ interaction }) => {
+ * 		//your code here
+ *  }
+ * })
+ * ```
+ * @end
+ */
+
+/** Marked TODO's will be reconfigured with release of sern v4! **/
+import { type GuildMember, type PermissionResolvable, type TextChannel } from 'discord.js';
+import { type CommandType, CommandControlPlugin, controller } from '@sern/handler';
+
+export const permsToString = (...perms: PermissionResolvable[]) => {
+  return new PermissionsBitField(perms)
+    .toArray()
+    .map((perm) => `\`${perm}\``)
+    .join(", ");
+};
+
+export function subcommandPermCheck(opts: Options) {
+  return CommandControlPlugin<CommandType.Slash>(async (ctx, [, args]) => {
```

Comment:
might be a good idea to not use `[,args]`. Will be removed in favor of SDT in version 4. consider using `ctx.options`

Reply:
This was going to be fixed when v4 was released but I fixed it!